### PR TITLE
Refactor bump_version script

### PR DESCRIPTION
**Pull request by masayag**
- reuse bump version logic
- add INFO messages
- update main chart version

### CODE REVIEW

**Review by gabriel-farache — APPROVED** [review]
```
dependency_version=$(awk -v dep="$dependency_name" '
            /^ *- name: / { in_dep = ($3 == dep) }
            in_dep && /^ *version:/ { gsub(/"/, "", $2); print $2; exit }
            ' ${main_chart_file})
``` 
is hard to get but I'll approve

**Comment by masayag** [comment]
> ```
> dependency_version=$(awk -v dep="$dependency_name" '
>             /^ *- name: / { in_dep = ($3 == dep) }
>             in_dep && /^ *version:/ { gsub(/"/, "", $2); print $2; exit }
>             ' ${main_chart_file})
> ```
> 
> is hard to get but I'll approve

Sure, let's break down the awk command:

`awk -v dep="$dependency_name"`: This starts an awk command and defines a variable dep with the value of $dependency_name. The -v option is used to pass variables into the awk script.

`'`: This marks the beginning of the awk script.

`/^ *- name: / { in_dep = ($3 == dep) }`: This pattern matches lines that start with - name: followed by optional spaces. When such a line is found, it sets the variable in_dep to true if the third field ($3) is equal to the value of the dep variable ($dependency_name). This helps to identify if we are currently processing the block related to the specified dependency.

`in_dep && /^ *version:/ { gsub(/"/, "", $2); print $2; exit }`: This condition checks if we are inside the block for the specified dependency (in_dep is true) and if the line matches the pattern version: followed by optional spaces. If both conditions are met, it removes any double quotes from the second field ($2) using gsub(/"/, "", $2) and then prints the second field ($2). Finally, it exits the awk command, ensuring we only print the version of the specified dependency.

`}`: This marks the end of the awk script.

`${main_chart_file}`: Specifies the file to be processed by awk.

In summary, this awk command searches for the block related to the specified dependency ($dependency_name) in the main Chart.yaml file (${main_chart_file}). It then extracts and prints the version of that dependency, ensuring it only prints the version within the correct block.
